webapp/src/services/near-nep21-util.js

```diff
@@ -0,0 +1,325 @@
+import { Contract} from 'near-api-js'
+import getConfig from '../config'
+
+const nearConfig = getConfig(process.env.NODE_ENV || 'development')
+const attachNear1 = '300000000000000';
+const attachNear2 = '60000000000000000000000';
+
+export async function getBalanceNEP( contractName ) {
+
+  window.nep21 = await new Contract(
+    window.walletConnection.account(),
+    contractName ,
+    {
+      // View methods are read only
+      viewMethods: ['get_balance'],
+      // Change methods modify state but don't receive updated data
+      changeMethods: []
+    }
+  )
+
+  return await window.nep21.get_balance({ owner_id: window.walletConnection.getAccountId() });
+
+}
+
+export async function incAllowance( token ) {
+  window.nep21 = await new Contract(
+    window.walletConnection.account(),
+    token.address ,
+    {
+      // View methods are read only
+      viewMethods: [],
+      // Change methods modify state but don't receive updated data
+      changeMethods: ['inc_allowance']
+    }
+  )
+  
+  try {
+    await window.nep21.inc_allowance({ 
+      escrow_account_id: nearConfig.contractName, 
+      amount: token.amount},
+      attachNear1,
+      attachNear2
+      );
+     console.log("DONE"); 
+    return true;
+  } catch(error) {
+    return false;
+  }
+
+}
+
+export async function getAllowance( token ) {
+  const accountId = window.accountId;
+  console.log(accountId);
+  window.nep21 = await new Contract(
+    window.walletConnection.account(),
+    token.address ,
+    {
+      // View methods are read only
+      viewMethods: ['get_allowance'],
+      // Change methods modify state but don't receive updated data
+      changeMethods: []
+    }
+  )
+  const allowance = await window.nep21.get_allowance({
+    owner_id: accountId, 
+    escrow_account_id: nearConfig.contractName });
+  console.log('Allowance: ', allowance);
+  return allowance;  
+}
+
+export async function gasCheck() {
+  // Set default
+  const near_limit = 0.6;
+  const bal = (await window.walletConnection.account().getAccountBalance()).available / 1000000000000000000000000;
+  if ( bal > near_limit ) {
+    return true;
+  }
+  return false;
+}
+
+function cleanString( str ) {
+  let ok = true;
+  let res = "";
+  for(var i = 0; i < str.length; ++i) {
+    if( str[i] === '0' && ok) {
+      
+    }
+    else {
+      ok = false;
+      res += str[i];
+    }
+  }
+  let last = res.length - 1;
+  while( res[last] === '0' ) {
+    res = res.slice(0, res.length - 1);
+    last--;
+  }
+  return res;
+
+}
+
+function setAmount( value ) {
+  let ok = false;
+  let val = 24;
+  let res = "";
+  const amount = cleanString( value );
+  for(var x of amount) {
+    if(x === '.') {
+      ok = true;
+    }
+    else if (x >= '0' && x <= '9') {
+      if(ok)
+        val--;
+      res += x;
+    } else {
+      console.error("Error: Wrong Input");
+      return -1;
+    }
+  }
+  res = res + '0'.repeat(val);
+  return res;
```

Comment:
this is far from optimal.
```
let start = 0;
for (; start< str.length; ++start) {
  if (str[start] !== '0')  break;
}
let end = str.length - 1;
for (; end> start; ++end) {
  if (str[from] !== '0')  break;
}
return str.slice(start,end+1)
```
Please also rename to `trimZeros`.

Reply:
Btw, why do we remove zeros from end?

Reply:
i thought there maybe any inconsistency in data, cleaning will result in less errors and to be on safe side as i got many unexpected errors when integrating UI

Reply:
OK, but I still don't understand why do you want remove zeros from the end. `0012300` will become `123`. Shouldn't this be `12300`?